core/src/main/java/com/linecorp/armeria/client/AbstractHttpRequestHandler.java

```diff
@@ -233,6 +240,31 @@ final void writeHeaders(RequestHeaders headers) {
         encoder.writeHeaders(id, streamId(), merged, headersOnly, promise);
     }
 
+    private static boolean shouldExpect100ContinueHeader(RequestHeaders headers) {
+        // Skip checking protocol version since Armeria is not fully compatible with HTTP/1.0.
+        // We can assume that the version is always HTTP/1.1 or later.
+        return headers.contains(HttpHeaderNames.EXPECT, HttpHeaderValues.CONTINUE.toString());
+    }
+
+    final boolean handle100Continue(HttpStatus status) {
+        if (state != State.NEEDS_100_CONTINUE) {
+            return true;
+        }
+
+        if (status != HttpStatus.CONTINUE) {
```

Comment:
It seems like we should continue to send the request without expectation header if the status is 417
```
A client that receives a [417 (Expectation Failed)](https://datatracker.ietf.org/doc/html/rfc9110#status.417) status code in response to a request containing a 100-continue expectation SHOULD repeat that request without a 100-continue expectation, since the 417 response merely indicates that the response chain does not support expectations (e.g., it passes through an HTTP/1.0 server).
```
https://datatracker.ietf.org/doc/html/rfc9110#section-10.1.1-11.4
Could you fix that?